### ISSUE TITLE
smard.de integration not working on sunday

### ISSUE THREAD

**Issue by aSauerwein** [mampfes/ha_epex_spot#46]
Hi, 

I'm using your integration with smard.de data source for region AT and relized that the sensor is 'unvailable' on synday afternoon.
After some debugging I found out, that the timestamps always show data from Sunday till Sunday midnight. 
an example:
we get all timestamps from this url:
https://www.smard.de/app/chart_data/4170/DE/index_hour.json
the latest timestamp as of now is: 1696197600000, Oct 1, 2023, 10:00:00 PM UTC ( 00:00 in CEST ) 

results in this series url:
https://www.smard.de/app/chart_data/4170/DE/4170_DE_hour_1696197600000.json
which gives us data from 1696197600000 ( Oct 1, 2023, 10:00:00 PM )  till 1696798800000 ( Oct 8, 2023, 9:00:00 PM). 
and that's why the check for `marketdata_now` fails:
https://github.com/mampfes/ha_epex_spot/blob/7b30c65e0f0fd6c6ceb1409cd2c1af1ff95e62b7/custom_components/epex_spot/SourceShell.py#L98

tl;dr:
on Sunday noon a new series for the net week is released, after which the service becomes unavailable. 

I think the right solution would be to not just use the latest timestamp https://github.com/mampfes/ha_epex_spot/blob/7b30c65e0f0fd6c6ceb1409cd2c1af1ff95e62b7/custom_components/epex_spot/EPEXSpot/SMARD/__init__.py#L101

we could choose the one that includes datetime.now(), or always get the last 2 and filter out "old" data before return. 

let me now what you think, and I'll create a pull request

**Comment by mampfes** [comment]
Hi, I create a fix in the main branch. Could you please test it (on Sunday)?

Please note that the main branch contains a major rewrite, therefore some more testing is required before I will release a new version. Please also note that the entity ids changed because they now follow the latest Home Assistant naming conventions.

**Comment by aSauerwein** [comment]
Hi!

Thanks for the quick fix. I just updated the integration. So far everything looks fine, will update you on Sunday

**Comment by mampfes** [comment]
Sorry, I missed that: Thanks for the detailed analysis!

**Comment by aSauerwein** [comment]
haven't looked closer yet. But it's still not working on Sunday. Will check again next Sunday

**Comment by mampfes** [comment]
I also configured SMARD with AT for testing. And today (Sunday) I couldn't reproduce the problem. Data was available the whole day. I'm using version 2.0.0.

**Comment by aSauerwein** [comment]
okay, interesting. must be a me problem then..
<img width="554" alt="image" src="https://github.com/mampfes/ha_epex_spot/assets/5446813/8a7498bd-7035-4dd3-b4ec-fcb5d01a6ffc">

**Comment by mampfes** [comment]
Did you install version 2.0.0?

**Comment by aSauerwein** [comment]
now it's working. 
maybe i had some files left from previous troubleshooting/debugging. 

Thanks again !
